java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiColumnValue.java

```diff
@@ -134,12 +135,22 @@ public void unpackMap(List<ColumnValue> keys, List<ColumnValue> values) {
     public void unpackStruct(List<Integer> structFieldIndex, List<ColumnValue> values) {
         StructObjectInspector inspector = (StructObjectInspector) fieldInspector;
         List<? extends StructField> fields = inspector.getAllStructFieldRefs();
+        // if we don't do any prune, we preserve all fields.
+        if (structFieldIndex == null) {
+            structFieldIndex = new ArrayList<>(fields.size());
+            for (int i = 0; i < fields.size(); i++) {
+                structFieldIndex.add(i);
+            }
+        }
```

Comment:
place this logic to `pruneOnStructSelectedFields`

Reply:
Updated.